fhir-audit/src/main/java/com/ibm/fhir/audit/logging/impl/WhcAuditCadfLogService.java

```diff
@@ -157,10 +157,10 @@ public void logEntry(AuditLogEntry logEntry) throws Exception {
         final String METHODNAME = "logEntry";
         logger.entering(CLASSNAME, METHODNAME);
 
-        // skip healthcheck and other bad operations
+        // skip healthcheck operation
         if (logEntry == null || logEntry.getContext() == null
-                || logEntry.getContext().getOperationName() == null
-                || logEntry.getContext().getOperationName().compareToIgnoreCase(HEALTHCHECKOP) == 0) {
+                || (logEntry.getContext().getOperationName() != null
+                && logEntry.getContext().getOperationName().compareToIgnoreCase(HEALTHCHECKOP) == 0)) {
```

Comment:
definitely just a style preference, but maybe change to equalsIgnoreCase while we're touching it?
```suggestion
                && logEntry.getContext().getOperationName().equalsIgnoreCase(HEALTHCHECKOP))) {
```